public/component_types/other/search_request.tsx

```diff
@@ -23,4 +24,68 @@ export class SearchRequest extends BaseComponent {
       },
     ];
   }
+
+  /**
+   * Validates if the index supports Knn if it is Knn query
+   * @param query The current search query
+   * @param indexSettings The settings of the selected index
+   * @returns Object containing validation result and warning message if needed
+   */
+  validateKnnQueryToHaveValidKnnIndex(
+    query: string,
+    indexSettings: string
+  ): {
+    isValid: boolean;
+    warningMessage?: string;
+  } {
+    try {
+      const queryString =
+        typeof query === 'string' ? query : JSON.stringify(query);
+      const hasKnnQuery = this.isKnnQuery(queryString);
+
+      if (hasKnnQuery) {
+        const isKnnEnabled = isKnnIndex(indexSettings);
+
+        if (!isKnnEnabled) {
+          return {
+            isValid: false,
+            warningMessage:
+              'Warning: You are using a neural/KNN query, but the selected index does not have KNN enabled (index.knn:true). Please select a KNN-enabled index for proper vector search functionality.',
+          };
+        }
+      }
+
+      return { isValid: true };
+    } catch (error) {
+      console.error('Error validating KNN query:', error);
+      return { isValid: true };
+    }
+  }
+
+  /**
+   * Determines if a query is a KNN query
+   * @param queryString The query string to check
+   * @returns boolean indicating if it's a KNN query
+   */
+  isKnnQuery(queryString: string): boolean {
+    try {
+      const query =
+        typeof queryString === 'string' ? JSON.parse(queryString) : queryString;
+
+      const queryAsString = JSON.stringify(query);
+
+      const hasKnn = queryAsString.includes('"knn":{');
+      const hasNeural = queryAsString.includes('"neural":{');
```

Comment:
How about leveraging the existing util fn `getFieldValue()` on the parsed query to check if the `knn` or `neural` property exists? This would be a cleaner way than depending on a string pattern, and it wouldn't require any string parsing on line 75.